inbm/dispatcher-agent/tests/unit/test_ota_parser.py

```diff
@@ -17,26 +17,32 @@ def setUp(self) -> None:
         self.parsed = MockXmlFunctions(fake_ota_invalid, is_file=False,
                                        schema_location=TEST_SCHEMA_LOCATION)
 
-    def test_parse_aota_a(self) -> None:
+    def test_parse_aota_with_empty_command_should_include_signature(self) -> None:
         result = AotaParser('remote').parse(
             {'cmd': '', 'app': '', 'fetch': 'https://www.google.com/'}, {}, self.parsed)
         self.assertTrue('signature' in result)
 
-    def test_parse_aota_b(self) -> None:
+    def test_parse_aota_valid_command_should_contain_fetch(self) -> None:
         result = AotaParser('remote').parse(
             {'cmd': 'cmd_1', 'app': 'app_1', 'fetch': 'http://www.google.com/'}, {}, self.parsed)
         self.assertTrue('google' in str(result))
+    
+    def test_parse_aota_with_signature_returns_correct_signature(self) -> None:
+        result = AotaParser('remote').parse(
+            {'cmd': 'cmd_1', 'app': 'app_1', 'fetch': 'http://www.google.com/', 'signature': 'abcdefg', 'sigversion': '384'}, {}, self.parsed)
+        self.assertEqual(384, result['hash_algorithm'])
+        self.assertEqual('abcdefg', result['signature'])
 
-    def test_parse_sota_a(self) -> None:
+    def test_parse_sota_empty_fields_should_include_hash_algorithm(self) -> None:
         result = SotaParser('remote').parse(
             {'cmd': '', 'signature': '', 'release_date': '', 'fetch': 'https://www.google.com/'}, {}, self.parsed)
         self.assertTrue('hash_algorithm' in result)
 
-    def test_parse_pota_a(self) -> None:
+    def test_parse_pota_empty_fields_should_include_fota(self) -> None:
         p = PotaParser('remote')
         result = p.parse(
             {'fota': '', 'sota': '', 'targetType': 'node', 'targets': ' ', 'fetch': 'https://www.google.com/'}, {}, self.parsed)
-        self.assertTrue('fota' in result)
+        self.assertTrue('fota' in result)    
```

Comment:
**Suggestion:** In 'test_parse_pota_empty_fields_should_include_fota', verify that 'fota' is not just present but also has the correct type or value. [enhancement]
**NOTE:** Ensure that the suggested code is comprehensive. This serves as an example of the code you could use. The user should verify it after accepting the suggestion.
```suggestion
    def test_parse_pota_empty_fields_should_include_fota(self) -> None:
        p = PotaParser('remote')
        result = p.parse(
            {'fota': '', 'sota': '', 'targetType': 'node', 'targets': ' ', 'fetch': 'https://www.google.com/'}, {}, self.parsed)
        self.assertIn('fota', result)
        self.assertIsNone(result.get('fota'))
```

Reply:
out of scope for this PR